### ISSUE TITLE
onClick is not working inside a popover with React 17.0.1

### ISSUE THREAD

**Issue by facuesr** [DevExpress/devextreme-react#473]
# Bug Report

**Package versions:**

    devexteme version: 20.2.3
    devextreme-react version: 20.2.3

**Steps to reproduce:**

**Current behavior:**

event onClick is not working inside a popover with react 17.0.1

example https://codesandbox.io/s/dazzling-silence-p96zs?file=/src/App.js

**Expected behavior:**

**Screenshots:**

**Comment by mrparag94** [comment]
how did you install it.. in my case npm install devextreme@20.2 devextreme-react@20.2 --save --save-exact doesn't work at all in react 17.0

**Comment by Semigradsky** [comment]
@mrparag94 try `npm install devextreme@20.2 devextreme-react@20.2 --save --save-exact --legacy-peer-deps`

> In order to get unblocked and install your project in spite of peerDependencies conflicts, you can use the **--legacy-peer-dep**s flag at install time

https://blog.npmjs.org/post/626173315965468672/npm-v7-series-beta-release-and-semver-major

**Comment by thebetternewt** [comment]
We discovered this issue as well. For us, the main benefit of upgrading to React 17 was the developer experience with the new JSX transform and the use of Fast Refresh over React Hot Reload in order to preserve state while developing.

The react team actually backported support for JSX transform to 16.14.0 (which allows to remove the `React` default import in modules not using it):

[https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html)

 So, if the improved developer experience is what anyone is looking for, you can actually just make sure to upgrade `react` and `react-dom` to `^16.14.0` and update `react-scripts` to `^4.0.1` (assuming you're using `create-react-app`) which uses Fast Refresh.

You may also need to include an `eslintConfig` in your package.json:
```json
 "eslintConfig": {
    "extends": [
      "react-app"
    ]
  },
```

If you're using **TypeScript**, you may also have to make sure you're on the latest version (4.1.2 at the time of this writing).

Though it doesn't address the bug in the original issue, hopefully this is helpful to those looking for the improved development experience that seems to come with React 17.

**Comment by LazyLahtak** [comment]
Hi everyone,

We fixed this issue. The fix will be available in v21.1.3.
If you need to fix it in an older version of DevExtreme, you can use [React Portals](https://reactjs.org/docs/portals.html). Here is an [example](https://codesandbox.io/s/overview-devextreme-popover-forked-s5iot?file=/App.js).

If you have further questions, feel free to contact us.
